### PR TITLE
Post UI screenshots in PR comments

### LINKED ISSUE #32828: [$200 bounty] Comment on PRs with UI report
We have a CI job that generates a bunch of UI screenshots in various states: https://github.com/commaai/openpilot/blob/master/.github/workflows/selfdrive_tests.yaml#L289. The goal of this bounty is to get those screenshots posted as a comment to PRs.
* only comments if `selfdrive/ui/` was changed
* comment gets updated when new commits are pushed
* must be formatted nicely (e.g. doesn't take up the whole screen). maybe put them in a table?

Since GitHub doesn't allow directly uploaded pictures to comments, we'll need to push them to this repo I setup: https://github.com/commaai/ci-artifacts. The `CI_ARTIFACTS_DEPLOY_KEY` secret should allow pushing there

### PR DESCRIPTION
**Pull request by signed-long**
Resolves #32828

See tests here: https://github.com/signed-long/openpilot/pull/13
- Changes outside of `selfdrive/ui` -> no comment
- Changes in `selfdrive/ui`  -> new comment
- New changes in `selfdrive/ui`  -> update comment

### CODE REVIEW

File: .github/workflows/selfdrive_tests.yaml [hunk 1]
```diff
@@ -322,3 +322,53 @@ jobs:
...
+        if: ${{ github.event_name == 'pull_request' && steps.changed-files.outputs.any_changed == 'true' }}
+        working-directory: ${{ github.workspace }}/ci-artifacts
+        run: |
+          git checkout -b openpilot/pr-${{ github.event.pull_request.number }}
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+          sudo mv ${{ github.workspace }}/selfdrive/ui/tests/test_ui/report/screenshots/* .
+          git add .
+          git commit -m "screenshots for PR #${{ github.event.pull_request.number }}"
+          git push origin openpilot/pr-${{ github.event.pull_request.number }} --force
```
[the remark this answers is not in the document]
Comment (signed-long) [reply .github/workflows/selfdrive_tests.yaml:350]: This is pushing to the ci-artifacts repo, the branch is named 'openpilot/pr-<or #>'

[the remark this answers is not in the document]
Comment (signed-long) [reply .github/workflows/selfdrive_tests.yaml:350]: My thoughts were keep the master branch having just a readme for quick checkouts in the GHA workflows, then force push to these PR branches for openpilot, new-connect etc

[the remark this answers is not in the document]
Comment (maxime-desroches) [reply .github/workflows/selfdrive_tests.yaml:350]: seems good with me

**Comment by maxime-desroches** [comment]
@adeebshihadeh Do we care about keeping those images after the PRs are merged? Because we might want to add a cleanup strategy for the ci-artifact repo

**Comment by adeebshihadeh** [comment]
We only care to the extent that GitHub has a max repo size. It’d be nice if we had the artifacts around for at least a couple months.

**Comment by maxime-desroches** [comment]
Congrats on the bounty!

**Pull request merged**